src/test/regress/sql/aux_ao_rels_stat.sql

```diff
@@ -195,3 +195,184 @@ ORDER BY index;
 -- Drop AO and aux tables
 DROP TABLE ao_table;
 
+-- Test the fix: Adding data to a partitioned AO table using `COPY FROM`  
+-- does not change the `tupcount` in the segment table on the QD.
+CREATE OR REPLACE FUNCTION get_total_tupcount(table_oid oid)
+RETURNS bigint AS $$
+DECLARE
+    aoseg_table_schema text;
+    aoseg_table_name text;
+    tupcount_result bigint;
+BEGIN
+    -- Get the AO segment table name for the given partition
+    SELECT n.nspname, c.relname INTO aoseg_table_schema, aoseg_table_name
+    FROM pg_appendonly AS a
+        JOIN pg_class AS c ON c.oid = a.segrelid
+        JOIN pg_namespace AS n ON c.relnamespace = n.oid
+    WHERE a.relid = table_oid;
+
+    -- Sum tupcount from all segments
+    EXECUTE format('SELECT SUM(tupcount) FROM %I.%I',
+                    aoseg_table_schema, aoseg_table_name)
+        INTO tupcount_result;
+
+    RETURN tupcount_result;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION get_tupcounts_for_partitions(input_name text)
+RETURNS TABLE (table_name name, tupcount bigint)
+AS $$
+DECLARE
+    main_table_oid oid;
+    main_table_name name;
+    schema_name name;
+BEGIN
+    -- Determine oid, schema and name of input table
+	SELECT c.oid, c.relname, n.nspname INTO main_table_oid, main_table_name, schema_name
```

Comment:
Indented by a tab, while other lines are indented by spaces. (also the same on lines 235, 236, 246, 247)